sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/FluxByteBufferContent.java

```diff
@@ -89,7 +100,36 @@ public Flux<ByteBuffer> toFluxByteBuffer() {
 
     @Override
     public boolean isReplayable() {
-        return false;
+        return isReplayable;
+    }
+
+    @Override
+    public BinaryDataContent toReplayableContent() {
+        Flux<ByteBuffer> bufferedFlux = content
+            .map(buffer -> {
+                if (buffer.isDirect()) {
+                    // deep copy direct buffers, they might be pooled memory or mapped file.
+                    ByteBuffer copy = ByteBuffer.allocate(buffer.remaining());
+                    copy.put(buffer);
+                    copy.flip();
+                    return copy;
+                } else {
+                    // these are heap buffers, safe to retain.
```

Comment:
Technically, heap buffers can be pooled too but I think this is safe enough

Reply:
Or we can just assume replaying incurs a deep cost and just buffer everything

Reply:
You're right. It's better to just deep copy.